unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/CharacterPreview/CharacterPreviewController.cs

```diff
@@ -137,6 +137,30 @@ private async UniTask UpdateModelAsync(AvatarModel newModel, CancellationToken c
             catch (Exception e) when (e is not OperationCanceledException) { Debug.LogException(e); }
         }
 
+        public async UniTask<Texture2D> TakeBodySnapshotAsync()
+        {
+            global::DCL.Environment.i.platform.cullingController.Stop();
```

Comment:
is there any way to inject this dependency instead of implicity coupling the singleton?